ui/src/app/pages/analysis/runtime-panel.tsx

```diff
@@ -883,20 +1032,42 @@ export const RuntimePanel = fp.flow(
   if (currentRuntime === undefined) {
     return <Spinner style={{width: '100%', marginTop: '5rem'}}/>;
   }
-
+  const createStandardComputeRuntimeRequest = (runtime: RuntimeConfig) => {
+    // The logic here is tricky to be compatible
+    // Use gceConfig when PD feature is not enabled OR reduce PD size OR update an existing runtime with no PD attached.
+    // TODO(RW-): 'Update an existing runtime with no PD attached' will be cleaned up
+    // post launch PD when all existing running Runtime shutdown.
+    if (!enablePD || (!pdSizeReduced && runtimeExists) || runtimeExistsWithoutPD) {
```

Comment:
note that there is a difference between my comment(`pdSizeIncrease`) and your change (`!pdSizeReduced`):

!pdSizeReduced can also be PD not exist. Would that work here?

Reply:
```   
 if(!enablePD || (pdSizeIncrease && runtimeExists) || (runtimeExists && !pdExists)) {
      return {
        gceConfig: {
          machineType: runtime.machine.name,
          diskSize: runtime.diskSize
        }
      };
    } 
```
Because using the above `pdSizeIncrease` will go to the wrong branch when there's no change on pdSize (`pdSizeIncrease = false`) but having changes on machineType (cpu, memory). But `!pdSizeReduced` could cover it.

Reply:
`!pdSizeReduced && runtimeExists` will go to the branch when PD does not exist but runTimeExist